### PR TITLE
minimint -> fedimint

### PR DESCRIPTION
**Pull request by Kodylow**
Addresses #286. Renames everything from minimint -> fedimint. 

Included all the markdown links which breaks the references to github.com/fedimint/minimint but immediately after merging this PR elsirion can change the repo name and all the paths will work. 

Also changed the nix configs from minimint -> fedimint which is errors the nix builds relying on the minimint cachix naming, see the cachix errors in the CI.

I can adjust it to keep those links all live for now.

### CODE REVIEW

File: integrationtests/Cargo.toml [hunk 1]
```diff
@@ -15,16 +15,16 @@ itertools = "0.10.0"
...
+fedimint-ln = { path = "../modules/fedimint-ln" }
+fedimint-mint = { path = "../modules/fedimint-mint" }
+fedimint-wallet = { path = "../modules/fedimint-wallet" }
 mint-client = { path = "../client/client-lib" }
 rand = "0.6.5"
 serde = { version = "1.0.118", features = [ "derive" ] }
 tokio = { version = "1.0.1", features = ["full"] }
 tracing ="0.1.22"
 tracing-subscriber = { version = "0.3.1", features = [ "env-filter" ] }
-hbbft = { git = "https://github.com/fedimint/hbbft", branch = "minimint" }
```
Comment (Kodylow) [inline integrationtests/Cargo.toml:29]: Branch name to update

**Comment by justinmoon** [comment]
I just created `fedimint` branches (same as current `minimint` branches) for our forks of `hbbft` and `bitcoin_hashes` and that fixed the nix build CI failures.

Markdown failures should be fixed once we rename the repo. So I would suggest (@Maan2003's idea) that first we rename the repo, then re-run CI and markdown test should pass, then merge this PR. Obviously, we should do this in one sitting rather quickly.

**Comment by elsirion** [comment]
I hope you have a script that does the renaming (e.g. using `sed`), otherwise this will be rebase hell :sweat_smile: 

The way to go imo is to build said script and get it to pass CI, post it here, and once there are few other PRs open run and merge as soon as CI runs through.

**Comment by justinmoon** [comment]
I propose we do this in 2 steps:

1) Rename the repo. That does 80% of the job. I propose we try this early next week.
2) Rename all occurrences of "minimint" in the code. We should have a script to do this and merge when there aren't many outstanding PRs.

**Comment by Kodylow** [comment]
Script that does all the changes within each file and changes the filenames:

```
find . -type f -exec sed -i 's/minimint/fedimint/g ; s/MiniMint/FediMint/g ; s/Minimint/Fedimint/g; s/MINIMINT/FEDIMINT/g' {} +
for i in ./*; do mv "$i" "${i/minimint/fedimint}"; done
for i in ./modules/*; do mv "$i" "${i/minimint/fedimint}"; done
```

**Comment by prusnak** [comment]
> Script that does all the changes within each file and changes the filenames:

I suggest to use `s/MiniMint/Fedimint/g` to avoid some headaches :-)

**Comment by justinmoon** [comment]
> I suggest to use `s/MiniMint/Fedimint/g` to avoid some headaches :-)

Good catch!

**Comment by justinmoon** [comment]
All the CLI crates are named a little strangely: `mint-client`, `clientd`, and `mint-client-cli`. All our other crates will have a `fedimint-` prefix. Perhaps these should as well, e.g. `fedimint-client`, `fedimint-clientd` and `fedimint-cli`? Perhaps `ln-gateway` should become `fedimint-gateway`?

**Comment by dpc** [comment]
The main server should be `fedimintd`, IMO. It is a common convention (`bitcoind`, etc.) makes it more clear when on referes to it, that it's about the server/binary not the whole project and so on.

**Comment by dpc** [comment]
Redid in #429, I think.

**Pull request closed without merging**